src/rules/parameterPropertiesRule.ts

```diff
@@ -0,0 +1,196 @@
+        options: {
+            type: "array",
+            items: {
+                type: "string",
+                enum: [OPTION_ALWAYS, OPTION_NEVER],
+            },
+        },
+        optionExamples: [true, [true, OPTION_ALWAYS], [true, OPTION_NEVER]],
+        type: "functionality",
+        typescriptOnly: true,
+        hasFix: true,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_ILLEGAL_PARAM_PROP = "Usage of parameter properties has been disallowed.";
+    public static FAILURE_MISSING_PARAM_PROP = "Use parameter properties instead of assigning to members in the constructor body.";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithFunction(sourceFile, walk, this.parseOptions(this.ruleArguments[0]));
+    }
+
+    private parseOptions(args: any): Options {
+        if (args && typeof args === "string" && (args === OPTION_ALWAYS || args === OPTION_NEVER)) {
+            return { never: args === OPTION_NEVER };
+        }
+        return { never: false };
+    }
+}
+
+function walk(ctx: Lint.WalkContext<Options>) {
+    const checkConstructors = (node: ts.Node): void => {
+        if (isConstructorDeclaration(node) && node.parameters.length > 0) {
+            const params: NamedNode[] = node.parameters.map((p: ts.ParameterDeclaration) => ({
+                name: (p.name as ts.Identifier).text,
+                node: p,
+            }));
+
+            if (ctx.options.never) {
+                for (const param of params) {
+                    if (isParameterProperty(param.node as ts.ParameterDeclaration)) {
+                        ctx.addFailureAtNode(param.node, Rule.FAILURE_ILLEGAL_PARAM_PROP);
+                    }
+                }
+                return;
+            }
+
+            const assignments: NamedNode[] = node.body!.statements
+                .filter(
+                    (stmt: ts.Statement) =>
+                        isExpressionStatement(stmt) &&
+                        isBinaryExpression(stmt.expression) &&
+                        isIdentifier(stmt.expression.right) &&
+                        isPropertyAccessExpression(stmt.expression.left) &&
+                        stmt.expression.left.expression.kind === 99,
+                )
+                .map((stmt: ts.Statement) => ({
+                    name: (((stmt as ts.ExpressionStatement).expression as ts.BinaryExpression)
+                        .right as ts.Identifier).text,
+                    node: stmt as ts.ExpressionStatement,
+                }));
+
+            for (const assignment of assignments) {
+                if (params.some((p: NamedNode) => p.name === assignment.name)) {
+                    ctx.addFailureAtNode(
+                        assignment.node,
+                        Rule.FAILURE_MISSING_PARAM_PROP,
+                        getFix(
+                            params.find((p: NamedNode) => p.name === assignment.name)!
```

Comment:
Store the results of `find` before the `some` call above. Remove call to `some`.